libs/application/templates/inheritance-report/src/fields/ReportFieldsRepeater/index.tsx

```diff
@@ -0,0 +1,286 @@
+  const { fields, append, remove } = useFieldArray<any>({
+    name: id,
+  })
+
+  const { setValue } = useFormContext()
+  const { formatMessage } = useLocale()
+  const answersValues = getValueViaPath(
+    application.answers,
+    id,
+  ) as Array<object>
+
+  /* ------ Stocks ------ */
+  const [rateOfExchange, setRateOfExchange] = useState(0)
+  const [faceValue, setFaceValue] = useState(0)
+  const [index, setIndex] = useState('0')
+
+  /* ------ Heirs ------ */
+  const [percentage, setPercentage] = useState(0)
+  const [taxFreeInheritance, setTaxFreeInheritance] = useState(0)
+  const [inheritance, setInheritance] = useState(0)
+  const [taxableInheritance, setTaxableInheritance] = useState(0)
+  const [inheritanceTax, setInheritanceTax] = useState(0)
+
+  /* ------ Total ------ */
+  const answersValuesTotal = answersValues?.length
+    ? answersValues.reduce((a: number, o: any) => {
+        return a + Number(o[props.sumField])
+      }, 0)
+    : 0
+
+  const [valueArray, setValueArray] = useState<Array<number>>(
+    answersValues?.length
+      ? answersValues.map((v: any) => Number(v[props.sumField]))
+      : [],
+  )
+
+  const [total, setTotal] = useState(
+    answersValues?.length ? answersValuesTotal : 0,
+  )
+
+  const handleAddRepeaterFields = () => {
+    const values = props.fields.map((field: object) => {
+      return Object.values(field)[1]
+    })
+
+    const repeaterFields = values.reduce((acc: any, elem: any) => {
+      acc[elem] = ''
+      return acc
+    }, {})
+
+    append(repeaterFields)
+  }
+
+  /* ------ Set total value ------ */
+  useEffect(() => {
+    const addTotal = id.replace('data', 'total')
+    setValue(addTotal, total)
+  }, [id, total, setValue])
+
+  /* ------ Set stocks value ------ */
+  useEffect(() => {
+    setValue(`${index}.value`, String(faceValue * rateOfExchange))
+  }, [faceValue, rateOfExchange, setValue])
+
+  /* ------ Set heirs calculations ------ */
+  useEffect(() => {
+    setTaxFreeInheritance(Skattleysismörk * percentage)
+    setInheritance(15000000 * percentage)
+    setTaxableInheritance(inheritance - taxFreeInheritance)
+    setInheritanceTax((inheritance - taxFreeInheritance) * 0.1)
+
+    setValue(`${index}.taxFreeInheritance`, taxFreeInheritance)
+    setValue(`${index}.inheritance`, inheritance)
+    setValue(`${index}.taxableInheritance`, taxableInheritance)
+    setValue(`${index}.inheritanceTax`, inheritanceTax)
+  }, [
+    index,
+    percentage,
+    taxFreeInheritance,
+    inheritance,
+    taxableInheritance,
+    inheritanceTax,
+    setValue,
+  ])
+
+  /* ------ Set fields from external data (realEstate, vehicles) ------ */
+  useEffect(() => {
+    if (props.fromExternalData && fields.length === 0) {
+      append(
+        (application.externalData.syslumennOnEntry?.data as any).estate[
+          props.fromExternalData
+        ],
+      )
+    }
+  }, [props, fields, append])
+
+  const calculateTotal = (v: any, index: any) => {
```

Comment:
Can these variable names be more descriptive?

Also on L134 the variable `v` is declared but then there is a reducer on L147 that also uses the variable `v`, maybe name these different things so that there is no issue with variable scope/scope misunderstandings.